semantic-tags/src/Tags/Tagging/Precise.hs

```diff
@@ -61,7 +61,7 @@ runTagging source
   . runReader source
 
 firstLine :: Source -> Text
-firstLine = Text.stripEnd . Text.takeWhile (/= '\n') . toText . Source.take 180
+firstLine = Text.stripEnd . Text.take 180 . Text.takeWhile (/= '\n') . Source.toText
```

Comment:
1. Do we need to ensure that we’re only ever passed the `Source` corresponding to the node, rather than the whole source, to avoid _n_ conversions of the entire file to `Text`?

2. Should we instead take the node’s `Range` as well as the `Source` and slice it ourselves to avoid converting the entire file because the caller forgot?

Reply:
Yeah, that's a good idea... All callers I see are doing a slice beforehand so we might as well make that part of the API.